File: String/Easy/1678. Goal Parser Interpretation.go
package main

import (
	"fmt"
	"strings"
)

func main() {

	command := "G()(al)"
	result := interpret(command)
	fmt.Println(result)
}

func interpret(command string) string {
	str := strings.Split(command, "")
	ans := ""

	for i := 0; i < len(str); i++ {
		if str[i] == "G" {
			ans += str[i]
		} else if str[i] == "(" && str[i+1] == ")" {
			ans += "o"
		} else if str[i] == "(" && str[i+1] == "a" {
			ans += "al"
		}
	}

	return ans
}

File: Miscellaneous/Python/1678. Goal Parser Interpretation.py
class Solution:
    def interpret(self, command):
        result = ""

        for i in range(len(command)):
            if command[i] == "G":
                result += "G"
            elif command[i] == "(" and command[i+1] == ")":
                result += "o"
                i += 1
            elif command[i] == "(" and command[i+1] == "a":
                result += "al"
                i += 2
        
        return result

     

obj = Solution()
command = "G()(al)"
ans = obj.interpret(command)
print(ans)
        